### ISSUE TITLE
Specializations and residues

### ISSUE THREAD

**Issue by RieksJ** [AmpersandTarski/Ampersand#795]
I have checked in four files in the `ampersand-models` repo; the commit is entitled `Hawaii - Searching updated with specializations (for Stef and Bas to look at)` (commit 06eca56c dated 13 jun 2018 11:07). Both `.adl` files can be compiled, the `.pop` files contain a test population and are INCLUDEd as necessary.

File `SearchRegisterForRecord.adl` is an application that implements a registration of people, as well as an interface that allows to search for people in that registration given text that you type in a search form. It is a demonstration of how residues can help you if you want to search records. 

File `SearchRegisterWithSpecializations.adl` is a functionally similar application; however, it has defined a specialization: `CLASSIFY Man ISA NaturalPerson`, and a relation that is typical for men: `manAttribute :: Man * ManAttribute`. This application shows two different ways of searching:

- Example 1 shows how to implicitly select Men when you are searching.
- Example 2 shows how to select both Men and non-Men when you are searching.

Please compare lines 31 and 55 of file `SearchRegisterWithSpecializations.adl`, and then compile the file, run the prototype and play around in the interface. Can you confirm that the behaviour that is exhibited in the interface is in fact as you would have expected?

**Comment by stefjoosten** [comment]
Let me rephrase. In your example you have
~~~
CLASSIFY "Man"   ISA NaturalPerson

manAttribute :: Man * ManAttribute
sfAttribute :: SearchForm * ManAttribute [UNI]
~~~
Then you have the following expression, which ensures that only Men are selected
~~~
sfAttribute~\manAttribute~
~~~
and the following expression, which ensures that only NaturalPersons are selected
~~~
sfAttribute~\(I[NaturalPerson];manAttribute)~
~~~
By running your script, you have computed evidence of these observations.

Your question is: is this the intended semantics?
@RieksJ, Please confirm that this is your question. In the meantime, let me think about it.

**Comment by RieksJ** [comment]
*   Your question is: is this the intended semantics?
  *   @RieksJ<https://github.com/RieksJ>, Please confirm that this is your question. In the meantime, let me think about it.

Yes, that's my questions.
This message may contain information that is not intended for you. If you are not the addressee or if this message was sent to you by mistake, you are requested to inform the sender and delete the message. TNO accepts no liability for the content of this e-mail, for the manner in which you use it and for damage of any kind resulting from the risks inherent to the electronic transmission of messages.

**Comment by hanjoosten** [comment]
What is required to close this issue? That's unclear to me. Could someone update this issue so that gets clear? If not, please close it.
